### PR TITLE
Add clustering in the UI

### PR DESCRIPTION
**Pull request by dsmilkov**
https://github.com/lilacai/lilac/assets/2294279/049ea0ff-6a1a-4b64-b60a-3632c74f65b0

### CODE REVIEW

File: notebooks/unmaintained/Hermes.ipynb
Comment (nsthorat) [inline notebooks/unmaintained/Hermes.ipynb:1]: revert this?

File: lilac/data/clustering.py [hunk 1]
```diff
@@ -184,10 +190,16 @@ def cluster(
...
     if not temp_path_exists or overwrite:
       # Since input is a function, map over the dataset to make a temporary column with that text.
+      if task_info:
+        task_info.message = 'Extracting text from items'
       dataset.map(input, output_path=path, overwrite=overwrite)
 
   clusters_exists = schema.has_field(cluster_output_path)
   if not clusters_exists or overwrite:
+    if task_info:
+      task_info.message = 'Computing super clusters' if category else 'Computing clusters'
```
Comment (nsthorat) [inline lilac/data/clustering.py:200]: maybe "cluster categories" since that's how it shows in the UI

File: lilac/data/dataset_duckdb.py [hunk 2]
```diff
@@ -2827,9 +2827,12 @@ def cluster(
     topic_fn: Optional[TopicFn] = None,
     overwrite: bool = False,
     remote: bool = False,
+    task_id: Optional[TaskId] = None,
```
Comment (nsthorat) [inline lilac/data/dataset_duckdb.py:2830]: I think we should avoid taskid in API from here forward -- can we create one internally like we do for map?

File: web/blueprint/src/lib/components/ComputeClusterModal.svelte [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        overwrite: options.overwrite
+      }
+    ]);
+    close();
+  }
+</script>
+
+{#if options}
+  <ComposedModal open on:submit={submit} on:close={close}>
+    <ModalHeader title="Cluster" />
```
Comment (nsthorat) [inline web/blueprint/src/lib/components/ComputeClusterModal.svelte:56]: maybe "Computer clusters" to parallel the other two methods which cause long expensive tasks

File: web/blueprint/src/lib/components/ComputeClusterModal.svelte [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      <div class="max-w-2xl">
+        <FieldSelect
+          filter={f => f.dtype?.type === 'string'}
+          defaultPath={options.input}
+          bind:path={options.input}
+          labelText="Field"
+        />
+      </div>
+      <div class="mt-8">
+        <div class="label text-s mb-2 font-medium text-gray-700">Remote</div>
```
Comment (nsthorat) [inline web/blueprint/src/lib/components/ComputeClusterModal.svelte:67]: before the next release we should disable this button on some environment-related condition via authinfo / have some explanation of what this is (maybe say run via Lilac Garden in the smaller text)

**Review by nsthorat — APPROVED** [review]

File: lilac/data/clustering.py (hunk 1 above)
Comment (dsmilkov) [reply lilac/data/clustering.py:200]: done

File: lilac/data/dataset_duckdb.py (hunk 2 above)
Comment (dsmilkov) [reply lilac/data/dataset_duckdb.py:2830]: i was just mimicking what compute_signal in router_dataset_signals does, we could clean this up in follow up PR

File: web/blueprint/src/lib/components/ComputeClusterModal.svelte (hunk 3 above)
Comment (dsmilkov) [reply web/blueprint/src/lib/components/ComputeClusterModal.svelte:56]: done

File: notebooks/unmaintained/Hermes.ipynb
Comment (dsmilkov) [reply notebooks/unmaintained/Hermes.ipynb:1]: done

File: web/blueprint/src/lib/components/ComputeClusterModal.svelte (hunk 4 above)
Comment (dsmilkov) [reply web/blueprint/src/lib/components/ComputeClusterModal.svelte:72]: done

**Pull request merged**